vhost/src/vhost_user/frontend_req_handler.rs

```diff
@@ -428,6 +535,12 @@ mod tests {
             let res = handler.handle_request().unwrap();
             assert_eq!(res, 0);
             handler.handle_request().unwrap_err();
+            assert_eq!(handler.handle_request().unwrap(), 0);
+            assert_eq!(handler.handle_request().unwrap(), 1);
+            assert_eq!(handler.handle_request().unwrap(), 0);
+            assert_eq!(handler.handle_request().unwrap(), 1);
+            assert_eq!(handler.handle_request().unwrap(), 0);
+            assert_eq!(handler.handle_request().unwrap(), 1);
```

Comment:
What about adding a comment that these tests are for shared objects?

BTW I tried to change the value asserted and the test passes in any case, so I think we should `join()` it in some way to check properly. (not for this PR)

Reply:
I'll add the comment.

I would've sworn I saw assertion errors here. I'll try again.

Also, I can open a follow-up PR to improve the logic of this test (if I do not improve it for this PR already).

Reply:
> I'll add the comment.
> 
> I would've sworn I saw assertion errors here. I'll try again.

It could be that I did something wrong while testing. ;-)

> 
> Also, I can open a follow-up PR to improve the logic of this test (if I do not improve it for this PR already).

That would be great! Thanks!

Reply:
> It could be that I did something wrong while testing. ;-)

You did not, it indeed requires joining. Which makes sense. But I checked and the first `asser_eq` that was there for the `FS_*` messages catches the fail without joining, so that is what threw me off. I failed to understand the difference... But I can push the `join()` already for this commit.

Reply:
@aesteve-rh what about this? Will we fix in another PR?

Reply:
Sorry, I'm not following. I've added the `join()` and a comment before the assertions to separate them from the other. Now if the check fails the test will surely fail. What is there left to fix?

In any case, whatever is left here to do related with the shared object feature, I'd say better to fix it here. Otherwise, e.g., improving test infra in general, it can go in another PR.

Reply:
Ooops, I missed that. I suggest adding a changelog on every update, like for mail-based workflow.

Also, since it fixes a pre-existing issue in our tests, what about having a separate commit with an explanation?